AK/UUID.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2020, Liav A. <liavalb@hotmail.co.il>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <AK/ByteBuffer.h>
+#include <AK/StringView.h>
+#include <AK/Types.h>
+
+namespace AK {
+
+class UUID {
+public:
+    UUID();
+    explicit UUID(ByteBuffer);
+    explicit UUID(const String&);
+    explicit UUID(const StringView&);
+    explicit UUID(const char*);
+    ~UUID() { }
+
+    String as_string() const;
+    const ByteBuffer& as_bytebuffer() const;
+
+    bool operator==(const UUID&) const;
+    bool operator!=(const UUID& other) const { return !(*this == other); }
+    bool operator<=(const UUID&) const = delete;
+    bool operator>=(const UUID&) const = delete;
+    bool operator<(const UUID&) const = delete;
+    bool operator>(const UUID&) const = delete;
+
+    bool is_null() const;
+
+private:
+    void convert_string_view_to_uuid(const StringView&);
+    ByteBuffer m_uuid_buffer;
```

Comment:
A UUID is always going to be the same length (128 bits), so let's use an array instead of a heap-allocating `ByteBuffer` :)